src/lando/utils/landing_checks.py

```diff
@@ -79,69 +79,149 @@ def result(self) -> str | None:
+        re.compile("^.github/workflows"),
+    ]
+    override_commit_message = "DOT_GITHUB_OVERRIDE"
+    error_message = "GitHub workflows directory"
 
     @override
     @classmethod
     def name(cls) -> str:
-        return "PreventSymlinksCheck"
+        return "PreventDotGithubCheck"
 
     @override
     @classmethod
     def description(cls) -> str:
-        return "Check for symlinks introduced in the diff."
+        return "Prevent changes to GitHub workflows directory."
 
-    symlinked_files: list[str] = field(default_factory=list)
 
-    def next_diff(self, diff: dict):
-        modes = diff["modes"]
+@dataclass
+class PreventNSPRCheck(PreventPathCheckMixin, PatchCheck):
+    """Prevent changes to vendored NSPR directories."""
 
-        # Check the file mode on each file and ensure the file is not a symlink.
-        # `rs_parsepatch` has a `new` and `old` mode key, we are interested in
-        # only the newly introduced modes.
-        if "new" in modes and modes["new"] == SYMLINK_MODE:
-            self.symlinked_files.append(diff["filename"])
+    paths = [
+        re.compile("^nsprpub/"),
+    ]
+    override_commit_message = "UPGRADE_NSPR_RELEASE"
+    error_message = "vendored NSPR directories"
 
-    def result(self) -> str | None:
-        if self.symlinked_files:
-            return (
-                "Revision introduces symlinks in the files "
-                f"{wrap_filenames(self.symlinked_files)}."
-            )
+    @override
+    @classmethod
+    def name(cls) -> str:
+        return "PreventNSSPRCheck"
+
+    @override
+    @classmethod
+    def description(cls) -> str:
+        return "Prevent changes to vendored NSPR directories."
 
 
 @dataclass
-class TryTaskConfigCheck(PatchCheck):
-    """Check for `try_task_config.json` introduced in the diff."""
+class PreventNSSCheck(PreventPathCheckMixin, PatchCheck):
+    """Prevent changes to vendored NSS directories."""
+
+    paths = [
+        re.compile("^security/nss/"),
+    ]
+    override_commit_message = "UPGRADE_NSS_RELEASE"
+    error_message = "vendored NSS directories"
 
     @override
     @classmethod
     def name(cls) -> str:
-        return "TryTaskConfigCheck"
+        return "PreventNSSCheck"
 
     @override
     @classmethod
     def description(cls) -> str:
-        return "Check for `try_task_config.json` introduced in the diff."
-
-    includes_try_task_config: bool = False
-
-    def next_diff(self, diff: dict):
-        """Check each diff for the `try_task_config.json` file."""
-        if diff["filename"] == "try_task_config.json":
-            self.includes_try_task_config = True
-
-    def result(self) -> str | None:
-        """Return an error if the `try_task_config.json` was found."""
-        if self.includes_try_task_config:
-            return "Revision introduces the `try_task_config.json` file."
+        return "Prevent changes to vendored NSS directories."
 
 
 @dataclass
 class PreventNSPRNSSCheck(PatchCheck):
```

Comment:
WDYT about splitting this into two separate checks instead of having one check which covers both NSPR and NSS? These two checks were combined in one Mercurial hook and ported to a single Lando check here, but it seems like it would be simpler to have separate checks for each (especially around the error message building).